src/wasm/kotlin/java/arcs/Addressable.kt

```diff
@@ -0,0 +1,30 @@
+package arcs
+
+/**
+ * Any object implementing this interface can be converted into a stable identifier. This
+ * could a platform specific pointer, identity hashcode, or id as long as it is guaranteed unique.
+ */
+interface Addressable
```

Comment:
Is the idea to use this on JVM too? I don't really understand its purpose if it's not just for wasm.

Reply:
It can be used for JVM if needed to map non-parcelable objects across process boundaries (ala PEC marshalling), but the real goal just to have a pure-JVM impl that doesn't require WASM specific forking. We could do as Jason suggests, which is to just implement an Any.toAddress() extension function inside of the /wasm directory, and then only use it in WasmRuntimeClient.  I'm open to nixing this, although I do like making it explicit as part of the type, in case we later need to migrate back to passing around raw WASM heap pointers. The base interface could then act like the old WasmObject.

Reply:
Implemented Jason's suggestion (which was clarified by Ray); now passing the build and test phases.